vmdb/app/helpers/api_helper/normalizer.rb

```diff
@@ -4,21 +4,32 @@ module Normalizer
     # Object or Hash Normalizer
     #
 
-    def normalize_hash(type, obj)
+    # Note: revisit merging direct and virtual normalize hash methods here once support for
+    # virtual subcollections is added.
+
+    def normalize_direct(type, name, obj)
+      return normalize_direct_array(type, name, obj) if obj.kind_of?(Array)
+      return normalize_hash(type, obj) if obj.respond_to?(:attributes) || obj.respond_to?(:keys)
+      normalize_attr_byname(type, name, obj)
+    end
+
+    def normalize_direct_array(type, name, obj)
+      obj.collect { |item| normalize_direct(type, name, item) }
+    end
```

Comment:
Which of these methods are part of the Normalizer API?  All of them?  Perhaps make the ones that are implementation details private.